File: go/problem#3208.go
package main

func numberOfAlternatingGroups(colors []int, k int) int {
	n := len(colors)
	colors = append(colors, colors...)
	count := 0
	l, r := 0, 1
	for ; r < n+k; r++ {
		if r-l == k {
			count++
			l++
		}
		if colors[r] == colors[r-1] {
			l = r
		}
	}
	return count
}

File: c/problems/problem#3208.c
#include <stdlib.h>

int numberOfAlternatingGroups(int *colors, int n, int k) {
  colors = realloc(colors, (n + k) * sizeof(int));
  for (int i = 0; i < k; i++) {
    colors[n + i] = colors[i];
  }
  int count = 0;
  int l = 0, r = 1;
  while (r < n + k) {
    if (r - l == k) {
      count++;
      l++;
    }
    if (colors[r] == colors[r - 1]) {
      l = r;
    }
    r++;
  }
  free(colors);
  return count;
}